src/main/java/uk/gov/companieshouse/confirmationstatementapi/service/ConfirmationStatementService.java

```diff
@@ -193,20 +199,35 @@ public ValidationStatusResponse isValid(String submissionId) throws SubmissionNo
                 validationStatus.setValidationStatusError(errors);
             }
 
-
             return validationStatus;
         } else  {
-            throw new SubmissionNotFoundException(
-                String.format("Could not find submission data for submission %s", submissionId));
+            throw new SubmissionNotFoundException(String.format("Could not find submission data for submission %s", submissionId));
         }
     }
 
     private boolean isConfirmed(SectionDataJson sectionData) {
-        return sectionData != null &&
+        return (sectionData != null) &&
                 (sectionData.getSectionStatus() == SectionStatus.CONFIRMED ||
                  sectionData.getSectionStatus() == SectionStatus.RECENT_FILING);
     }
 
+    private boolean isConfirmed(RegisteredEmailAddressDataJson sectionData, LocalDate madeUpToDate) {
+        if (isEcctEnabled(madeUpToDate)) {
+            return (sectionData != null) &&
+                    (sectionData.getSectionStatus() == SectionStatus.CONFIRMED ||
+                    sectionData.getSectionStatus() == SectionStatus.RECENT_FILING ||
+                    sectionData.getSectionStatus() == SectionStatus.INITIAL_FILING);
+        }
+
+        return true;
+    }
+
+    private boolean isEcctEnabled(LocalDate madeUpToDate) {
+        var ecctStartDate = LocalDate.parse(ecctStartDateStr, DATE_TIME_FORMATTER);
+
+        return isBeforeOrEqual(madeUpToDate, ecctStartDate);
```

Comment:
madeupdate should be on or after ecctstartdate. Maybe Javadoc to clarify this logic.

Reply:
Still looks wrong

Reply:
I agree it is wrong, apologies for my confusion! I need to look at the unit test for this logic...

Reply:
It looks a bit strange but it is actually correct, the parameter order of isBeforeOrEqual may be confusing.  I have added a Javadoc for that method in a new commit to clarify. (I have also checked how the original author of that method was using it, see call on line 215:

```
isBeforeOrEqual(localDateNow.get(), submissionData.getMadeUpToDate());
```